### PR TITLE
fix: fix regression in NLog local decoration when logging objects as parameters

### PR DESCRIPTION
**Pull request by nr-ahemsath**
## Description

See bug #1459.  PR #1393 improved NLog local decoration coverage, but broke logging of objects as parameters, e.g.:
`_logger.LogDebug("Start Request: {@request}", request);`

This PR fixes the issue and adds test coverage for the parameterized logging case.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)
- [ ] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs [hunk 1]
```diff
@@ -77,6 +91,11 @@ private void DecorateLogMessage(object logEvent, Type logEventType, IAgent agent
                 return;
             }
 
+            if (formattedMessage.Contains("NR-LINKING"))
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs:94]: Maybe we could make this a function in `LoggingHelpers` so we don't need to know about the token here?

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs [hunk 2]
```diff
@@ -69,6 +70,19 @@ private void DecorateLogMessage(object logEvent, Type logEventType, IAgent agent
             {
                 return;
             }
+            var formattedMetadata = LoggingHelpers.GetFormattedLinkingMetadata(agent);
+
+            // This wrapper for NLog uses a belt-and-suspenders approach to decorating log output. We first try to decorate the Message property,
```
Comment (jaffinito) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs:75]: Like that we called this out.

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj [hunk 3]
```diff
@@ -70,9 +70,9 @@
 
     <!-- MySql.Data .NET/Core references -->
     <PackageReference Include="MySql.Data" Version="6.10.7" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
-    <PackageReference Include ="MySql.Data" Version="8.0.15" Condition="'$(TargetFramework)' == 'net5.0'" />
-    <PackageReference Include ="MySql.Data" Version="8.0.30" Condition="'$(TargetFramework)' == 'net6.0'" />
-    <PackageReference Include ="MySql.Data" Version="8.0.32.1" Condition="'$(TargetFramework)' == 'net7.0'" />
+    <PackageReference Include="MySql.Data" Version="8.0.15" Condition="'$(TargetFramework)' == 'net5.0'" />
```
Comment (jaffinito) [inline tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj:73]: Were these changes intentional?

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs (hunk 1 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/NLogLogging/NLogWrapper.cs:94]: That seems like it wouldn't be a bad idea,

File: tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj (hunk 3 above)
Comment (nr-ahemsath) [reply tests/Agent/IntegrationTests/SharedApplications/Common/MultiFunctionApplicationHelpers/MultiFunctionApplicationHelpers.csproj:73]: Not intentional, but they're just whitespace changes that VS did automatically.  The pattern in the rest of the file is for there to be no space between `Include` and `=` so I think we should do it.

**Review by chynesNR — APPROVED** [review]
Looks good!

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Logging/LoggingHelpers.cs [hunk 4]
```diff
@@ -50,7 +51,12 @@ public static string GetFormattedLinkingMetadata(IAgent agent)
 
             // This is a positional blob so we want the delimiters left in when no data is  present.
             // NR-LINKING|{entity.guid}|{hostname}|{trace.id}|{span.id}|
-            return "NR-LINKING|" + entityGuid + "|" + hostname + "|" + traceId + "|" + spanId + "|" + entityName + "|";
+            return LinkingToken + "|" + entityGuid + "|" + hostname + "|" + traceId + "|" + spanId + "|" + entityName + "|";
+        }
+
+        public static bool ContainsLinkingToken(string text)
```
Comment (chynesNR) [inline src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Logging/LoggingHelpers.cs:57]: Very clean!

**Review by jaffinito — APPROVED** [review]
LGTM!

**Review by chynesNR — APPROVED** [review]

**Review by JcolemanNR — APPROVED** [review]
Looks good! I'm glad to see we were able to identify and fix the issue in basic (non web) NLog.
